packages/vgrammar/src/types/theme.ts

```diff
@@ -1,3 +1,115 @@
-export interface ThemeSpec {
+import type {
+  IArc3dGraphicAttribute,
+  IArcGraphicAttribute,
+  IAreaGraphicAttribute,
+  ICircleGraphicAttribute,
+  IColor,
+  IGroupGraphicAttribute,
+  IImageGraphicAttribute,
+  ILineGraphicAttribute,
+  IPathGraphicAttribute,
+  IPolygonGraphicAttribute,
+  IPyramid3dGraphicAttribute,
+  IRect3dGraphicAttribute,
+  IRectGraphicAttribute,
+  IRichTextGraphicAttribute,
+  ISymbolAttribute,
+  ISymbolGraphicAttribute,
+  ITextGraphicAttribute
+} from '@visactor/vrender';
+import type {
+  ArcLabelAttrs,
+  CircleAxisAttributes,
+  CircleCrosshairAttrs,
+  ColorLegendAttributes,
+  ContinuousPlayerAttributes,
+  DataLabelAttrs,
+  DataZoomAttributes,
+  DiscreteLegendAttrs,
+  DiscretePlayerAttributes,
+  LineAxisAttributes,
+  LineCrosshairAttrs,
+  LineLabelAttrs,
+  PolygonCrosshairAttrs,
+  RectCrosshairAttrs,
+  RectLabelAttrs,
+  SectorCrosshairAttrs,
+  SizeLegendAttributes,
+  SliderAttributes,
+  SymbolLabelAttrs,
+  TooltipAttributes
+} from '@visactor/vrender-components';
+import type { IPadding } from '@visactor/vutils';
+import type { RecursivePartial } from './base';
+
+export interface IMarkTheme {
+  rect?: IRectGraphicAttribute & { y1?: number; x1?: number };
```

Comment:
this type seems the same to `BasicEncoderSpecMap`, maybe you can use `BasicEncoderSpecMap` directly

Reply:
Fixed.